### PR TITLE
#54730 (Resources > Pool: Edit temp project for user)

### PR DESCRIPTION
**Pull request by namdh99**
expect: gọi api update (check xem có api update temp ko, ko có thì tạo) 
actual: gọi api AddUserFromPoolToTempProject -> sai

### CODE REVIEW

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/b5763172-5466-45fa-b771-c5e39590c73c

File: aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs [hunk 1]
```diff
@@ -248,5 +248,13 @@ public async Task<List<RetroReviewInternHistoriesDto>> GetRetroReviewInternHisto
         {
             return await _resourceManager.GetRetroReviewInternHistories(input.Emails);
         }
+
+        [HttpPost]
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs:253]:
```
        [AbpAuthorize(PermissionNames.Resource_TabPool_AddTempProject, PermissionNames.Resource_TabPool_EditTempProject)]
        public async Task UpdateTempProjectForUser(AddResourceToProjectDto input)
        {
            var allowUpdateTempProjectForUser = await PermissionChecker.IsGrantedAsync(PermissionNames.Resource_TabPool_EditTempProject);
            await _resourceManager.CreatePresentProjectUserAndNofity(input, allowUpdateTempProjectForUser);
        }
```

=>

```
        [PermissionNames.Resource_TabPool_EditTempProject)]
        public async Task UpdateTempProjectForUser(AddResourceToProjectDto input)
        {            
            await _resourceManager.CreatePresentProjectUserAndNofity(input, true);
        }
```

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/049a6226-20a4-4421-8f38-5c7469cbf59e

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/plan-resource/add-user-to-temp-project-dialog/add-user-to-temp-project-dialog.component.ts [hunk 2]
```diff
@@ -43,8 +46,7 @@ export class AddUserToTempProjectDialogComponent extends AppComponentBase implem
       isPool: true,
       projectRole: this.user.projectRole
     }
-    console.log(this.user)
-    this.resourceService.AddUserToTempProject(requestBody).pipe(catchError(this.resourceService.handleError)).subscribe(rs=>{
+    this.resourceService.updateTempProjectForUser(requestBody).pipe(catchError(this.resourceService.handleError)).subscribe(rs=>{
       abp.notify.success("Add successful")
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/plan-resource/add-user-to-temp-project-dialog/add-user-to-temp-project-dialog.component.ts]: abp.notify.success("Add successful")
=> abp.notify.success("Update successful")

File: aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs [hunk 3]
```diff
@@ -47,7 +49,8 @@ public class ResourceAppService : ProjectManagementAppServiceBase
             UserManager userManager,
             ResourceManager resourceManager,
             ISettingManager settingManager,
-            IUserAppService userAppService)
+            IUserAppService userAppService,
+            IWorkScope workScope)
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs]: remove IWorkScope workScope
ko được thêm workscope vào đây
viết thêm function UpdateTempProjectForUser ở ResourceManager

**Review by tiennguyenhuuncc — COMMENTED** [review]
check lại comment

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/565d7f39-74e0-413c-9f44-2138813bf422

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/85a91564-09c5-493a-877a-e27a93711bea

**Review by tiennguyenhuuncc — APPROVED** [review]

File: aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs [hunk 4]
```diff
@@ -248,5 +249,11 @@ public async Task<List<RetroReviewInternHistoriesDto>> GetRetroReviewInternHisto
         {
             return await _resourceManager.GetRetroReviewInternHistories(input.Emails);
         }
+
+        [HttpPost]
+        public async Task UpdateTempProjectForUser(UpdateTempProjectForUserDto input)
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/Resource/ResourceAppService.cs:254]: ko gắn permission vào api à?

**Review by tiennguyenhuuncc — COMMENTED** [review]
check comment

**Pull request merged**